language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/syntaxhighlighter/BallerinaSyntaxHighlightService.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ballerinalang.langserver.extensions.ballerina.syntaxhighlighter;
+
+import org.eclipse.lsp4j.jsonrpc.services.JsonRequest;
+import org.eclipse.lsp4j.jsonrpc.services.JsonSegment;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * An extension interface for Language server to fetch information about ballerina syntax hightlighter.
+ *
+ * @since 0.981.2
+ */
+@JsonSegment("ballerinaSyntaxHighlighter")
+public interface BallerinaSyntaxHighlightService {
+
+    @JsonRequest
+    CompletableFuture<BallerinaSynResponse> list(BallerinaSynRequest request);
```

Comment:
Let's move this into a `@JsonNotification` as discussed. And we can publish `server->client` notification in [here](https://github.com/ballerina-platform/ballerina-lang/blob/aa5aaa0fb327e984f6e843942c7a096f95acd28e/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/diagnostic/DiagnosticsHelper.java#L86). For example; `client.publishTextHighlighting(...)`. You can implement `publishTextHighlighting(...)` into [ExtendedLanguageClient.java](https://github.com/ballerina-platform/ballerina-lang/blob/master/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/client/ExtendedLanguageClient.java#L27:L33)